iceoryx_posh/include/iceoryx_posh/runtime/service_discovery.hpp

```diff
@@ -78,15 +70,20 @@ class ServiceDiscovery
     void invalidateTrigger(const uint64_t uniqueTriggerId);
     iox::popo::WaitSetIsConditionSatisfiedCallback
     getCallbackForIsStateConditionSatisfied(const popo::SubscriberState state);
-    roudi::ServiceRegistry m_serviceRegistry;
+
+    // use dynamic memory to reduce stack usage
+    /// @todo #1155 improve solution to avoid stack usage without using dynamic memory
+    std::unique_ptr<roudi::ServiceRegistry> m_serviceRegistry{new iox::roudi::ServiceRegistry};
```

Comment:
Nitpick: Once `m_deleter` is a `cxx::function` we could use the `cxx::unique_ptr` here and reduce the dependency to the STL. Not necessary as it's a temporary solution anyway.

Reply:
Yes, but as it is now I would prefer `std::unique_ptr` which properly deals with dynamic memory.